apps/web/app/[locale]/(app)/dashboard/[organization]/[list]/table.tsx

```diff
@@ -2,25 +2,77 @@
 
 import { DataTable } from '@/components/data-table/data-table';
 import { Attribute, Contact } from '@prisma/client';
-import { getContactColumns } from './columns';
+import { TableProps, getContactColumns } from './columns';
 import { CustomerAttributes } from '@/lib/types/data-columns';
+import { HeaderActions } from './header-actions';
+import { useToast } from '@/components/ui/use-toast';
+import { Table } from '@tanstack/react-table';
+import { useRouter } from 'next/navigation';
 
 type ContactTableProps = {
   contacts: Contact[];
   attributes: Attribute[];
+  organization: string;
+  list: string;
 };
 
-export const ContactTable = ({ contacts, attributes }: ContactTableProps) => {
-  const columns = getContactColumns(attributes);
+export const ContactTable = ({
+  contacts,
+  attributes,
+  organization,
+  list,
+}: ContactTableProps) => {
+  const { toast } = useToast();
+  const router = useRouter();
+
+  const deleteContacts = async (table: Table<TableProps>) => {
+    // get all selected contacts
+    const selectedRows = table.getFilteredSelectedRowModel().rows;
+    const contactIds = selectedRows.map(row => row.original.id);
+
+    // send request to delete contacts
+    const response = await fetch('/api/contacts', {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ contactIds }),
+    });
+
+    if (response.ok) {
+      const data = await response.json();
+      toast({
+        title: 'Contacts deleted',
+        description: `${data.deletedContacts} contacts were successfully deleted`,
+      });
+      table.resetRowSelection();
+      // Fixme: We should probably use other revalidation strategies
+      router.refresh();
```

Comment:
Something like this? https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#on-demand-revalidation

Reply:
Thought of using SWR, but you are totally right, using Next.js built-in features is probably sufficient 😄 I restructured it a bit and moved it back to the server action as an API call isn't really necessary. Let me know what you think about it 🙂

Reply:
Looks great 🙌

Reply:
@marvin-splitt btw, I recently saw this video: https://youtu.be/9jqPbNgzHOA?si=-2Pl3Dw0ei5s7bRg&t=2097

might be another valid approach if we want to reestablish refetching/revalidation logic on the client in future scenarios.